src/main/org/bson/ByteBuf.java

```diff
@@ -101,7 +101,12 @@
   /**
    * The size of an unsigned 32-bit integer: 2^32
    */
-  private static long UINT32_SIZE = (long)Math.pow(2, 32);
+  private static long UINT32_SIZE = 4294967296L;
+
+  /**
+   * The max of an signed 32-bit integer: 2^31
+   */
+  private static long INT32_MAX = 2147483648L;
```

Comment:
I think this should be `2147483647L`, and it seems like there needs to be a test added that would fail with `2147483648L` being used instead of `2147483647L`.

Reply:
the test shouldn't fail if it's 2147483648L, being that it puts an unsigned integer, 2^31 can fit there... I put a test there to verify that my code correctly converts 2^31 into something that can fit into a signed int